gadget/gadget.go

```diff
@@ -591,7 +593,7 @@ func validateRole(vs *VolumeStructure, vol *Volume) error {
 		if vs.Filesystem != "" && vs.Filesystem != "none" {
 			return errors.New("mbr structures must not specify a file system")
 		}
-	case SystemBoot, "":
+	case SystemBoot, BootImage, BootSelect, "":
```

Comment:
do we need a test here?

Reply:
and also added test for BootImage and BootSelect roles